pyppeteer/jshandle.py

```diff
@@ -0,0 +1,613 @@
+    ):
+        super().__init__(context, client, remoteObject)
+        self._page = page
+        self._frameManager = frameManager
+        self._disposed = False
+
+    def asElement(self) -> Optional['ElementHandle']:
+        return self
+
+    async def contentFrame(self):
+        nodeInfo = await self._client.send(
+            'DOM.describeNode',
+            {
+                'objectId': self._remoteObject.get('objectId')
+            }
+        )
+        frameId = nodeInfo.get('node', {}).get('frameId')
+        if not isinstance(frameId, str):
+            return
+        return self._frameManager.frame(frameId)
+
+    async def _scrollIntoViewIfNeeded(self):
+        error = await self.evaluate(
+            """
+            async(element, pageJavascriptEnabled) => {
+              if (!element.isConnected)
+                return 'Node is detached from document';
+              if (element.nodeType !== Node.ELEMENT_NODE)
+                return 'Node is not of type HTMLElement';
+              // force-scroll if page's javascript is disabled.
+              if (!pageJavascriptEnabled) {
+                element.scrollIntoView({block: 'center', inline: 'center', behavior: 'instant'});
+                return false;
+              }
+              const visibleRatio = await new Promise(resolve => {
+                const observer = new IntersectionObserver(entries => {
+                  resolve(entries[0].intersectionRatio);
+                  observer.disconnect();
+                });
+                observer.observe(element);
+              });
+              if (visibleRatio !== 1.0)
+                element.scrollIntoView({block: 'center', inline: 'center', behavior: 'instant'});
+              return false;
+            }
+            """,
+            self._page._javascriptEnabled
+        )
+        if error:
+            raise BrowserError(error)
+
+    async def _clickablePoint(self):
+        result, layoutMetrics = await asyncio.gather(
+            self._client.send(
+                'DOM.getContentQuads',
+                {'objectId': self._remoteObject['objectId']}
+            ),
+            self._client.send('Page.getLayoutMetrics')
+        )
+        if not result or not result.get('quads', {}).get('length'):
+            raise BrowserError('Node is either not visible or not an HTMLEelement')
+        clientWidth, clientHeight = layoutMetrics.layoutViewport
+        quads = [
+            self._fromProtocolQuad(quad)
+            for quad in self._intersectQuadWithViewport(quad, clientWidth, clientHeight)
+            if computedQuadArea(quad) > 1
+        ]
+        if not quads:
+            raise BrowserError('Node is either not visible or not an HTMLElement')
+        quad = quads[0]
+        x = 0
+        y = 0
+        for point in quad:
+            x += point.x
+            y += point.y
+        return {
+            'x': x / 4,
+            'y': y / 4,
+        }
+
+    def _getBoxModel(self):
```

Comment:
Shouldn't this method be asynchronous?

Reply:
yeah it should be, weirdly enough it's not async in puppeteer though it's referenced and used as async method -- I'll attribute this to js weirdness and move on :D